packages/react-native/React/Fabric/Mounting/ComponentViews/Text/RCTParagraphComponentView.mm

```diff
@@ -252,27 +252,27 @@ - (void)disableContextMenu
 - (void)handleLongPress:(UILongPressGestureRecognizer *)gesture
 {
 #if !TARGET_OS_UIKITFORMAC
-  if (@available(iOS 16.0, *)) {
+  if (@available(iOS 16.0, macCatalyst 16.0, *)) {
     CGPoint location = [gesture locationInView:self];
     UIEditMenuConfiguration *config = [UIEditMenuConfiguration configurationWithIdentifier:nil sourcePoint:location];
     if (_editMenuInteraction) {
       [_editMenuInteraction presentEditMenuWithConfiguration:config];
-    }
-    return;
-  }
-  // TODO: Adopt showMenuFromRect (necessary for UIKitForMac)
-  UIMenuController *menuController = [UIMenuController sharedMenuController];
+	// [visionOS
+	} else {
+	  UIMenuController *menuController = [UIMenuController sharedMenuController];
 
-  if (menuController.isMenuVisible) {
-    return;
-  }
+	  if (menuController.isMenuVisible) {
+		return;
+	  }
 
-  if (!self.isFirstResponder) {
-    [self becomeFirstResponder];
-  }
+	  if (!self.isFirstResponder) {
+		[self becomeFirstResponder];
+	  }
```

Comment:
This if-block isn't present upstream.

Reply:
Similar comment, may have unintentionally been removed upstream by my change: https://github.com/facebook/react-native/pull/42277